watch/funcs.go

```diff
@@ -155,7 +170,7 @@ func serviceWatch(params map[string]interface{}) (WatcherFunc, error) {
 		health := p.client.Health()
 		opts := makeQueryOptionsWithContext(p, stale)
 		defer p.cancelFunc()
-		nodes, meta, err := health.Service(service, tag, passingOnly, &opts)
+		nodes, meta, err := health.ServiceMultipleTags(service, tags, passingOnly, &opts)
```

Comment:
One issue here is that this transparently breaks compatibility with older versions of Consul.

There are a few ways to address this but I think one easy way is to use the `health.Service` method when `tag != ""` and tags originally was nil. Basically if the watcher is using the tags array it would implicitly opt-in to using the newer incompatible APIs.

Reply:
will do

Reply:
@mkeeler sorry I misunderstood your objection at first.

I do not believe this actually is a BC break given that the current implementation of the `Health` service itself does this:
```golang
// Service is used to query health information along with service info
// for a given service. It can optionally do server-side filtering on a tag
// or nodes with passing health checks only.
func (h *Health) Service(service, tag string, passingOnly bool, q *QueryOptions) ([]*ServiceEntry, *QueryMeta, error) {
	var tags []string
	if tag != "" {
		tags = []string{tag}
	}
	return h.service(service, tags, passingOnly, q, false)
}
```

The underlying `service` method expects `tags []string`, not `tag string`.  Additionally, given that the end result is `?tag=tag1&tag=tag2&tag=tag3`, I do not think its possible for this to be an actual BC break.

Of course, please correct me if I am wrong...

Reply:
@mkeeler: Could you please elaborate the BC component to my PR?  I don't believe it actually does.  I have not had a chance to spin up a old 0.6.4 agent or something to test against, but I can spend some time later today doing so and provide the logs if it will help move this PR along.

Reply:
@dcarbone After digging a little deeper, I think you are right. The API and HTTP layer of Consul just use multiple `tag` params to specify a multi-tag filter. Its when this gets translated into RPCs by the HTTP server where things could go poorly but that is already handled and not something you need to worry about with this PR.